experimental/dds/tree2/src/feature-libraries/simple-tree/proxyNodeCursor.ts

```diff
@@ -0,0 +1,193 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/core-utils";
+
+import {
+	EmptyKey,
+	rootFieldKey,
+	type FieldKey,
+	type TreeTypeSet,
+	type Value,
+	CursorLocationType,
+} from "../../core";
+// eslint-disable-next-line import/no-internal-modules
+import { leaf } from "../../domains/leafDomain";
+import {
+	type ContextuallyTypedNodeData,
+	type TreeDataContext,
+	getPossibleTypes,
+	isFluidHandle,
+	isPrimitiveValue,
+} from "../contextuallyTyped";
+import {
+	type CursorAdapter,
+	type CursorWithNode,
+	stackTreeNodeCursor,
+	stackTreeFieldCursor,
+} from "../treeCursorUtils";
+import { TreeNodeSchema } from "../typed-schema";
+import { type ProxyNode } from "./types";
+/**
+ * Creates a {@link CursorAdapter} capable of mapping proxy-domain trees to the editable tree domain.
+ *
+ * @param context - The {@link TreeEntity.context} of the parent node under which the tree is being inserted.
+ * @param typeSet - The types allowed under the parent node under which the tree is being inserted.
+ *
+ * @privateRemarks
+ * This is exported solely for testing purposes.
+ * Consumers should use {@link cursorFromProxyTree} instead.
+ */
+export function createProxyTreeAdapter<TNode extends ProxyNode<TreeNodeSchema, "javaScript">>(
```

Comment:
I don't think making this generic adds any value. Sadly I don't think the proxy API provides any generic tree reading friendly types, and its not covariant over schema, so there isn't really a well-formed equivalent to FlexTreeNode.

Anyway, this generic setup is also wrong, since the adapter below assumes all children of a node are the same type as the node.

Something like `type TreeEntity = TreeValue | { readonly [P in string]?: TreeEntity } | ReadonlyMap<string|TreeEntity>` is the type you want I guess?

The lack of read only APIs preventing covariance over schema, and lack of type safety for proxy objects vs other objects makes thus such a mess.